grpc4bmi/bmi_client_singularity.py

```diff
@@ -29,58 +27,160 @@ class BmiClientSingularity(BmiClient):
     The client picks a random port and expects the container to run the server on that port.
     The port is passed to the container using the BMI_PORT environment variable.
 
-    >>> from grpc4bmi.bmi_client_singularity import BmiClientSingularity
-    >>> image = 'docker://ewatercycle/wflow-grpc4bmi:latest'
-    >>> client = BmiClientSingularity(image, input_dir='wflow_rhine_sbm', output_dir='wflow_output')
-    >>> client.initialize('wflow_rhine_sbm/wflow_sbm.ini')
-    >>> client.update_until(client.get_end_time())
-    >>> del client
-
     Args:
         image: Singularity image. For Docker Hub image use `docker://*`.
-        input_dir (str): Directory for input files of model
-        output_dir (str): Directory for input files of model
-        timeout (int): Seconds to wait for gRPC client to connect to server
-        delay (int): Seconds to wait for Singularity container to startup, before connecting to it
-        extra_volumes (Dict[str,str]): Extra volumes to attach to Singularity container.
+        input_dirs (Iterable[str]): Directories for input files of model.
+
+            All of directories will be mounted read-only inside Singularity container on same path as outside container.
 
-            The key is the hosts path and the value the mounted volume inside the container.
-            Contrary to Docker client, extra volumes are always read/write
+        work_dir (str): Working directory for model.
+
+            Directory is mounted inside container and changed into.
+
+            To create a random work directory you could use
 
-            For example:
-            
             .. code-block:: python
-            
-                    {'/data/shared/forcings/': /data/forcings'}
-    """
-    INPUT_MOUNT_POINT = "/data/input"
-    OUTPUT_MOUNT_POINT = "/data/output"
 
-    def __init__(self, image, input_dir=None, output_dir=None, timeout=None,
-                 delay=0, extra_volumes=None):
+              from tempfile import TemporaryDirectory
+              from grpc4bmi.bmi_client_singularity import BmiClientSingularity
+
+              work_dir = TemporaryDirectory()
+
+              image = 'docker://ewatercycle/marrmot-grpc4bmi:latest'
+              client =  BmiClientSingularity(image, work_dir.name)
```

Comment:
I am wondering if it is possible to run BmiClientSingularity() with a docker image as it is defined here. Why not define the image as `ewatercycle-marrmot-grpc4bmi.sif`?

Reply:
It is possible, as it used in https://github.com/eWaterCycle/grpc4bmi/blob/41277060a2fe7a51309c93089987a4dd6d772ed4/test/test_singularity.py#L13-L18 . 

I can use *.sif files in examples, but I will also add the build command so the file does not appear from nowhere.